atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -439,6 +439,7 @@ protected void checkGetPreconditions(TableReference tableRef) {
                     rawBuilder.put(result);
                 }
                 Map<Cell, Value> raw = rawBuilder.build();
+                validateExternalAndCommitLocksIfNecessary(tableRef);
```

Comment:
And yep, I think the ordering at 587 is ok, but I moved this up above the test to `return endOfData()`, since we might get an empty batch, but should fail if the locks have already expired.